src/Phpstan/Rule/EnforceStrictMocking.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Lendable\PHPUnitExtensions\Phpstan\Rule;
+
+use Lendable\PHPUnitExtensions\StrictMocking as StrictMockingTrait;
+use Lendable\PHPUnitExtensions\TestCase as StrictMockingTestCase;
+use PhpParser\Node;
+use PhpParser\Node\Name;
+use PhpParser\Node\Stmt\Class_;
+use PHPStan\Analyser\Scope;
+use PHPStan\Reflection\ClassReflection;
+use PHPStan\Rules\Rule;
+use PHPStan\Rules\RuleErrorBuilder;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * @implements Rule<Class_>
+ */
+final class EnforceStrictMocking implements Rule
+{
+    /**
+     * @var array<class-string, int>
+     */
+    private readonly array $pardoned;
+
+    /**
+     * @param list<class-string> $pardoned
+     */
+    public function __construct(array $pardoned)
+    {
+        $this->pardoned = \array_flip($pardoned);
+    }
+
+    public function getNodeType(): string
+    {
+        return Class_::class;
+    }
+
+    public function processNode(Node $node, Scope $scope): array
```

Comment:
We probably want to ignore abstract classes too here as we only should be enforcing this at the concrete test case level?

Reply:
https://github.com/Lendable/phpunit-extensions/pull/14 implements this